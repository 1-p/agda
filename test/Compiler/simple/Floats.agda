
module _ where

open import Common.Prelude

print : Float → IO Unit
print x = putStrLn (primShowFloat x)

printB : Bool → IO Unit
printB true  = putStrLn "true"
printB false = putStrLn "false"

_/_ = primFloatDiv
_==_ = primFloatEquality
_<_ = primFloatLess

NaN : Float
NaN = 0.0 / 0.0

Inf : Float
Inf = 1.0 / 0.0

-Inf : Float
-Inf = -1.0 / 0.0

main : IO Unit
main =
  putStr "NaN   = " ,, print NaN   ,,
  putStr "Inf   = " ,, print Inf   ,,
  putStr "-Inf  = " ,, print -Inf  ,,
  -- Disabled due to #1856 and #1857
  -- putStr "√2    = " ,, print (primFloatTimes 2.0 (primSin (primFloatDiv pi 4.0))) ,,
  -- putStr "e     = " ,, print (primExp 1.0) ,,
  putStr "Inf == Inf  = " ,, printB (Inf == Inf) ,,
  putStr "NaN < -Inf  = " ,, printB (NaN < -Inf) ,,
  return unit
