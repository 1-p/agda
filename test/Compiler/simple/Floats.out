EXECUTED_PROGRAM

ret > ExitSuccess
out > 123.0 = 123.0
out > -42.0 = -42.0
out > NaN   = NaN
out > Inf   = Infinity
out > -Inf  = -Infinity
out > -0.0  = -0.0
out > NaN == NaN  = true
out > Inf == Inf  = true
out > NaN < -Inf  = false
out > NaN < -5.0  = true
out > 0.0 == -0.0 = false
out >
